feature: introduce upgrade command

sreis opened the pull request:
Close #325

This PR adds the upgrade command to the opstrace CLI, which required a couple of refactorings:
* Adds build info to its own package to use that information between the different packages consistently.
* Moves functions to fetch the cluster kubeconfig to "./util" to share code between the commands.
* Moves functions to find the region to "./util" to share code between the commands.

sreis commented:
I'm opening the PR now to check if I didn't break the tests.

sreis commented on packages/controller/src/tasks/general.ts:
```diff
@@ -34,5 +41,26 @@ export function* deployControllerResources(config: {
+      break;
+    }
+
+    case ControllerResourcesDeploymentStrategy.Update: {
+      log.debug("updating controller resources");
+      yield all([resources.map(updateResource)]);
+      break;
+    }
```
This is called but the resource is never updated...

```
2021-02-09T18:48:00.562Z debug: updating controller resources
2021-02-09T18:48:00.563Z debug: updateResource for hasura-admin-secret(ns: kube-system):
2021-02-09T18:48:00.563Z debug: updateResource for opstrace-controller(ns: kube-system):
2021-02-09T18:48:00.564Z debug: updateResource for opstrace-controller(ns: kube-system):
2021-02-09T18:48:00.564Z debug: updateResource for opstrace-controller-clusteradmin-binding(ns: default):
2021-02-09T18:48:00.565Z info: controller deployment found in informer-based state
```

jgehrcke commented:
@sreis thanks for sharing early -- let me know when you'd like some feedback : ) 
Great progress. Feels good!

sreis commented:
@jgehrcke Currently dealing with yarn dependency errors in a clean checkout. Have a look at the PR when you have spare time. Any review is more than welcome.

sreis commented:
Fixed the build under a clean checkout and learned a bit about `tsconfig.json`. 

Fixed the issue with the controller resources not getting updated. The resource update request was not being sent because it's not waiting for the Promise to conclude. This is the same code pattern used in the installer and uninstaller. I'm not yet familiar with redux-saga to fully understand the issue. I worked around it by checking if the current deployment has been updated (this is done by waiting for the deployment status observed generation to change).

triclambert commented:
Nit: update the copyright date.

sreis commented:
Rebased with main. Going to merge when tests are green and follow up with additional PRs in case of post-merge reviews.
